File: osmToolsBasic/src/main/java/com/asamm/osmTools/cmdCommands/CmdShp2osm.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.asamm.osmTools.cmdCommands;

import com.asamm.osmTools.Parameters;
import com.asamm.osmTools.mapConfig.ItemMap;

import java.io.File;

/**
 *
 * @author volda
 */
public class CmdShp2osm extends Cmd{

    String input;
    String output;

    public CmdShp2osm(ItemMap map, String input, String output) {
        super(map, ExternalApp.NO_EXTERNAL_APP);
        this.input = input;
        this.output = output;

        //test if shp2osm.py scripot exist
        if (!new File (Parameters.getShp2osmDir()).exists()){
            throw new IllegalArgumentException ("Shp2Osm script in location" + Parameters.getShp2osmDir() +"  does not exist!");
        }
    }
    
    public void createCmd(){
        addCommand(Parameters.getPythonDir());
        addCommand(Parameters.getShp2osmDir());
        addCommand("--id");
        addCommand(String.valueOf(Parameters.costlineBorderId));
        addCommand("--output");
        addCommand(output);
        addCommand(input);
    }
}

File: osmToolsBasic/src/main/java/com/asamm/osmTools/cmdCommands/CmdShp2osm.kt
package com.asamm.osmTools.cmdCommands

import com.asamm.osmTools.config.AppConfig
import java.nio.file.Path
import kotlin.io.path.absolutePathString

class CmdShp2osm : Cmd(ExternalApp.LOMAPS_TOOLS) {

    fun shp2osm(input: Path, output: Path) {
        builder()
            .add("shp2osm")
            .add("--id", AppConfig.config.coastlineConfig.nodeBorderId++.toString())
            .add("--input", input.absolutePathString())
            .add("--output", output.absolutePathString())
            .execute()
    }
}